src/OpenSpaceToolkit/Physics/Environment/Atmospheric/Earth.cpp

```diff
@@ -253,13 +308,59 @@ Unique<Earth::Impl> Earth::ImplFromType(
     }
     else if (aType == Earth::Type::Exponential)
     {
-        return std::make_unique<ExponentialImpl>(aType, anEarthFrameSPtr, anEarthRadius, anEarthFlattening);
+        return std::make_unique<ExponentialImpl>(
+            aType, anInputDataSourceType, anEarthFrameSPtr, anEarthRadius, anEarthFlattening
+        );
     }
     else if (aType == Earth::Type::NRLMSISE00)
     {
-        return std::make_unique<NRLMSISE00Impl>(
-            aType, anEarthFrameSPtr, anEarthRadius, anEarthFlattening, aSunCelestialSPtr
-        );
+        if (anInputDataSourceType == Earth::InputDataSourceType::Undefined)
+        {
+            throw ostk::core::error::runtime::Undefined("Input data source type");
+        }
+        else if (anInputDataSourceType == Earth::InputDataSourceType::ConstantFluxAndGeoMag)
+        {
+            if (!aF107ConstantValue.isDefined())
+            {
+                throw ostk::core::error::runtime::Undefined("F10.7 constant value");
+            }
+            else if (!aF107AConstantValue.isDefined())
+            {
+                throw ostk::core::error::runtime::Undefined("F10.7a constant value");
+            }
+            else if (!aKpConstantValue.isDefined())
+            {
+                throw ostk::core::error::runtime::Undefined("Kp constant value");
+            }
+
+            return std::make_unique<NRLMSISE00Impl>(
+                aType,
+                anInputDataSourceType,
+                NRLMSISE00::InputDataSourceType::ConstantFluxAndGeoMag,
+                aF107ConstantValue,
+                aF107AConstantValue,
+                aKpConstantValue,
+                anEarthFrameSPtr,
+                anEarthRadius,
+                anEarthFlattening,
+                aSunCelestialSPtr
+            );
+        }
+        else if (anInputDataSourceType == Earth::InputDataSourceType::SpaceWeatherFile)
+        {
+            return std::make_unique<NRLMSISE00Impl>(
+                aType,
+                anInputDataSourceType,
+                NRLMSISE00::InputDataSourceType::SpaceWeatherFile,
+                aF107ConstantValue,
+                aF107AConstantValue,
+                aKpConstantValue,
```

Comment:
```suggestion
```

mega nit: I think this is a little clearer that they are unused

Reply:
Yeah, we would need `Earth::NRLMSISE00Impl::NRLMSISE00Impl` to accept default args for this (and probably change the order of args though 🤔)